### PR TITLE
MEN-4737 Monitoring wrapper alignment

### PR DESCRIPTION
**Pull request by merlin-northern**
MEN-4737 Monitoring wrapper alignment

Goes with https://github.com/mendersoftware/monitor-client/pull/17
* include service file
* add a correct link to the mender-monitord

ChangeLog:title
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc [hunk 1]
```diff
@@ -23,4 +24,5 @@ do_install() {
         -C ${S} \
         DESTDIR=${D} \
         install
+    ln -sf ${datadir}/mender-monitor/mender-monitord ${D}${bindir}/mender-monitord
```
Comment (lluiscampos) [inline meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc:27]: Still pending discussion for this in https://github.com/mendersoftware/monitor-client/pull/17#discussion_r664089103

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Nice try, but in yocto (and more specifically our `meta-mender`) things are never that simple :P

The yocto way of dealing with systemd services is:
* Inherit `systemd` class: https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-mender/mender-configure/mender-configure.inc#L4
* Set the corresponding `SYSTEMD_SERVICE` variable: https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-mender/mender-configure/mender-configure.inc#L23
* Append the file only when using `mender-systemd` feature: https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-mender/mender-configure/mender-configure.inc#L19
* Have a dedicated snstall target for `mender-systemd` feature: https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-mender/mender-configure/mender-configure.inc#L59

**Comment by merlin-northern** [comment]
> Nice try, but in yocto (and more specifically our `meta-mender`) things are never that simple :P
> 
> The yocto way of dealing with systemd services is:
> 
> * Inherit `systemd` class: https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-mender/mender-configure/mender-configure.inc#L4
> * Set the corresponding `SYSTEMD_SERVICE` variable: https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-mender/mender-configure/mender-configure.inc#L23
> * Append the file only when using `mender-systemd` feature: https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-mender/mender-configure/mender-configure.inc#L19
> * Have a dedicated snstall target for `mender-systemd` feature: https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-mender/mender-configure/mender-configure.inc#L59

right! thanks. fixed. tested, works.

File: meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc [hunk 2]
```diff
@@ -18,9 +24,20 @@ do_version_check() {
 }
 addtask do_version_check after do_unpack before do_install
 
+SYSTEMD_SERVICE_${PN}_mender-systemd = "mender-monitor.service"
+
 do_install() {
     oe_runmake \
         -C ${S} \
         DESTDIR=${D} \
         install
```
Comment (lluiscampos) [inline meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc:33]: Does this `install` target also include `install-systemd`? Then we cannot use it and we need to specify all the other `install-*` targets here.

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Looking better :nerd_face:

File: meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc (hunk 2 above)
Comment (merlin-northern) [reply meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc:33]: I would prefer removing install-systemd from the Makefile from monitor-client -- it makes better sense, can it be like that?

Comment (lluiscampos) [reply meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc:33]: As you wish.

File: meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc (hunk 1 above)
Comment (lluiscampos) [reply meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc:34]: Discussion there is done, decided to keep the link.

However this is already done by the `install` target, so we should not need it here.

Comment (merlin-northern) [reply meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc]: discussed over gmeet. thanks!

File: meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc (hunk 2 above)
Comment (merlin-northern) [reply meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor.inc:33]: aligned with https://github.com/mendersoftware/monitor-client/pull/17

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**